### DISCUSSION TITLE
{closure}(): Argument #1 ($record) must be of type App\Models\NominalRoll, null given,

### DISCUSSION THREAD

**Discussion by IykeOkey** [filamentphp/filament#13275] [Help]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

When I run this code "->actions([
                Action::make('copyToSchedule')
                    ->label('Copy to Schedule')
                    ->action(function (Nomroll $record) {
                        // Perform the copy operation directly
                        Schedule::create($record->only(['field1', 'field2', 'field3'])); // specify the fields to copy
                    })
                    ->requiresConfirmation()
                    ->color('primary'),
            ]);"
I received the following TypeError;
App\Filament\Resources\RunSalaryResource::App\Filament\Resources\{closure}(): Argument #1 ($record) must be of type App\Models\NominalRoll, null given, called in C:\Users\USER\Payroll\Arabel\vendor\filament\support\src\Concerns\EvaluatesClosures.php.

Observing the following about action function: Filament\Actions\Concerns\HasAction::action takes @param \Closure|string|null $action

How can an action function take a database model as @param rather than a string?

**Comment by leandrocfe** [comment]
Where are you using this action?

**Comment by IykeOkey** [comment]
$table
            ->headerActions

**Reply by leandrocfe** [reply]
$record is available in the row actions, not in the header action
